great_expectations/checkpoint/checkpoint.py

```diff
@@ -109,6 +110,15 @@ class Config:
             Renderer: lambda r: r.serialize(),
         }
 
+    @override
+    def json(self, **kwargs: Any) -> str:
+        # Necessary override to check that all children validation definitions are added.
+        # JSON encoder will raise an error on the first non-added child.
+        added, errors = self.is_added()
+        if not added:
+            raise CheckpointRelatedResourcesNotAddedError(errors=errors)
+        return super().json(**kwargs)
```

Comment:
This method is used in the store to serialize - if we don't have this, we check the first validation definition and possibly fail.

Perhaps there is a better way to do this but this allows us to aggregate all errors quite easily.